src/ray/protobuf/autoscaler_state_service.proto

```diff
@@ -0,0 +1,123 @@
+syntax = "proto3";
+option cc_enable_arenas = true;
+
+package ray.rpc;
+
+import "src/ray/protobuf/intance_manager.proto";
+// ============= Cluster Resources ====================
+//
+// Following fields represents the Cluster Resources autoscaler interested
+// in.
+
+// Represents an anti-affinity constraint. A bundle with this constraint
+// can't be allocated to a node that has a label with the same name and
+// value.
+message AntiAffinityConstraint {
+  string label_name = 1;
+  string label_value = 2;
+}
+
+message PlacementConstraint {
+  oneof constraint {
```

Comment:
nit: should we just flatten this? `oneof` is a no-op over the wire so it tends to be confusing (it's not optimized like a C union).